tests/jballerina-unit-test/src/test/resources/test-src/jvm/type-test-expr.bal

```diff
@@ -131,3 +131,37 @@ function testAnyJsonTypes() returns [boolean, boolean, boolean, boolean] {
 
     return [aa is json, bb is json, dd is json, ff is json];
 }
+
+function testIsLikeForTupleWithRestDescriptor() {
```

Comment:
Shall we also add tests for where the source size is less than the target tuple member size?

Reply:
Added